pkg/controllers/management/restrictedadminrbac/cluster.go

```diff
@@ -0,0 +1,122 @@
+package restrictedadminrbac
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/go-multierror"
+	apisv3 "github.com/rancher/rancher/pkg/apis/management.cattle.io/v3"
+	v3 "github.com/rancher/rancher/pkg/apis/management.cattle.io/v3"
+	"github.com/rancher/rancher/pkg/rbac"
+	"github.com/rancher/wrangler/pkg/name"
+	"github.com/rancher/wrangler/pkg/relatedresource"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+// clusterOwnerSync ensures that the given enqueued GRB has a cluster-owner CRTB in all downstream clusters iff the role is GlobalRestrictedAdmin.
+func (r *rbaccontroller) clusterOwnerSync(key string, grb *v3.GlobalRoleBinding) (runtime.Object, error) {
+	if grb == nil || grb.DeletionTimestamp != nil || grb.GlobalRoleName != rbac.GlobalRestrictedAdmin {
+		return nil, nil
+	}
+	clusters, err := r.clusters.List(metav1.ListOptions{})
```

Comment:
would be better to add a ClusterLister to rbaccontroller

Reply:
We can also just pull the cache off directly - `clusters.Cache()`. 

Would using a cache here run into timing issues? IIRC the caches take a short time to update - is it possible that a cluster could cause the related resource to enqueue the GRB before the cache has a record of the cluster?

Reply:
Ended up going with the Wrangler cache since I try to use Wrangler over Norman when possible. As for the possible race condition with the cache,  in theory, the informer causing the event handler is the same informer used for the cache, and I'd rather start with a cache and update later if we are hitting this condition.